lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -416,7 +416,13 @@ bool isTppAdd(linalg::GenericOp linalgOp) {
     return false;
   if (!isBinaryOp(linalgOp))
     return false;
-  if (!allOperandsHaveSameShapeAndStrides(linalgOp->getOperands().getTypes()))
+  if (!allOperandsHaveSameShapeAndStrides(linalgOp->getOperands().getTypes()) &&
+      (verifyTppIdentityBroadcastingRules(linalgOp.getOperand(1).getType(),
```

Comment:
why `verifyTppIdentityBroadcastingRules`? You can just use `verifyBroadcastableShapeImpl` (see: https://github.com/plaidml/tpp-mlir/commit/e8002870ba5115d6885911b0eff0e24d8ce1f79d).